hacktheback/forms/models.py

```diff
@@ -0,0 +1,168 @@
+    `forms.Response` is a response to a specific form.
+    """
+
+    HACKER_APPLICANT = "HA"
+    MISCELLANEOUS = "MI"
+    TYPE_CHOICES = [
+        (HACKER_APPLICANT, "Hacker Applicant"),
+        (MISCELLANEOUS, "Miscellaneous"),
+    ]
+
+    title = models.CharField(max_length=128)
+    hackathon = models.ForeignKey(
+        Hackathon, on_delete=models.CASCADE, related_name="forms"
+    )
+    description = models.TextField()
+    type = models.CharField(
+        max_length=2, choices=TYPE_CHOICES, default=MISCELLANEOUS
+    )
+    is_draft = models.BooleanField(default=True)
+
+
+class Question(GenericModel, OrderedModel):
+    """
+    A question within a :model: `forms.Form`.
+
+    Each answered question in a :model: `forms.Response` for a :model:
+    `forms.Form` is an :model: `forms.Answer`.
+    """
+
+    SHORT_TEXT = "ST"
+    LONG_TEXT = "LT"
+    SELECT = "SL"
+    MULTISELECT = "MS"
+    HYPERLINK = "HL"
+    PHONE = "PH"
+    EMAIL = "EM"
+    RADIO = "RD"
+    FILE = "FL"
+    TYPE_CHOICES = [
+        (SHORT_TEXT, "Short Text"),
+        (LONG_TEXT, "Long Text"),
+        (SELECT, "Select"),
+        (HYPERLINK, "Hyperlink"),
+        (PHONE, "Phone"),
+        (EMAIL, "Email"),
+        (MULTISELECT, "Multiselect"),
+        (RADIO, "Radio"),
+        (FILE, "File"),
+    ]
+
+    OPTION_TYPES = [SELECT, MULTISELECT, RADIO]
+    SOLO_OPTION_TYPES = [SELECT, RADIO]
+
+    form = models.ForeignKey(
+        Form, on_delete=models.CASCADE, related_name="questions"
+    )
+    label = models.CharField(max_length=128)
+    type = models.CharField(
+        max_length=2, choices=TYPE_CHOICES, default=SHORT_TEXT
+    )
+    description = models.TextField(
+        null=True, help_text="A question's help text."
+    )
+    placeholder = models.CharField(
+        max_length=128,
+        null=True,
+        help_text="The value for a question's HTML placeholder.",
+    )
+    required = models.BooleanField(default=False)
+    default_answer = models.TextField(null=True)
+
+    order_with_respect_to = "form"
+
+    class Meta(OrderedModel.Meta):
+        unique_together = ["form", "label"]
+
+
+class QuestionOption(GenericModel, OrderedModel):
+    """
+    A selectable option for a :model: `forms.Question` that has an option
+    type (Select, Multiselect, Radio).
+    """
+
+    question = models.ForeignKey(
+        Question, on_delete=models.CASCADE, related_name="options"
+    )
+    label = models.CharField(max_length=128)
+    value = models.CharField(max_length=64)
+    default_answer = models.BooleanField(default=False)
+    # Edge case: If an admin deletes a QuestionOption but a related
+    # AnswerOption exists, don't delete it but instead set this to True.
+    persist_deletion = models.BooleanField(
+        default=False,
+        help_text="The option has been deleted and won't be valid for future responses.",
+    )
+
+    class Meta(OrderedModel.Meta):
+        unique_together = ["question", "label"]
```

Comment:
It's true we dont want question and label to be the same, but i think its more important that the value and question is not the same

Reply:
What's the difference between value and label again?

Reply:
it's similar to the enum we have where Radio = 'RD', tho would it be better if we just only had label as the value?